modules/caddytls/internalissuer.go

```diff
@@ -114,12 +119,16 @@ func (iss InternalIssuer) Issue(ctx context.Context, csr *x509.CertificateReques
 
 	// ensure issued certificate does not expire later than its issuer
 	lifetime := time.Duration(iss.Lifetime)
-	if time.Now().Add(lifetime).After(issuerCert.NotAfter) {
-		// TODO: log this
-		lifetime = issuerCert.NotAfter.Sub(time.Now())
+	if time.Now().Add(time.Duration(iss.Lifetime)).After(issuerCert.NotAfter) {
+		lifetime = time.Until(issuerCert.NotAfter)
+		iss.logger.Warn("cert lifetime would exceed issuer NotAfter, clamping lifetime",
+			zap.Duration("orig_lifetime", time.Duration(iss.Lifetime)),
+			zap.Duration("lifetime", lifetime),
+			zap.Time("not_after", issuerCert.NotAfter),
+		)
 	}
 
-	certChain, err := auth.Sign(csr, provisioner.SignOptions{}, customCertLifetime(iss.Lifetime))
+	certChain, err := auth.Sign(csr, provisioner.SignOptions{}, customCertLifetime(caddy.Duration(lifetime)))
```

Comment:
there was a bug here before, though I'm guessing the number of times the cert's lifetime would've needed to be clamped was pretty low 😉

Reply:
Great catch, thanks.